datanode/src/storage_interface.py

```diff
@@ -586,6 +629,58 @@ def delete_multi_operation(self, z, grids, uss_id, gufi):
       result = self._format_status_code_to_jsend(400, e.message)
     return result
 
+  def delete_uvr(self, z, grids, uvr):
+    """Updates multiple GridCells' metadata by removing the entry for the UVR.
+
+    Args:
+      z: zoom level in slippy tile format
+      grids: list of (x,y) tiles to delete
+      uvr: validated nested-dict UVR data structure
+    Returns:
+      JSend formatted response (https://labs.omniti.com/labs/jsend)
+    """
+    log.debug('Deleting UVR %s in %d grid cells for %s...',
+              uvr['message_id'], len(grids), uvr['originator_id'])
+    try:
+      status = 200
+      for _ in range(DELETE_ATTEMPTS):
+        # First, plan all deletions without modifying anything
+        deletions = []
+        for x, y in grids:
+          if not slippy_util.validate_slippy(z, x, y):
+            raise ValueError('Invalid slippy grids for lookup')
+          (content, metadata) = self._get_raw(z, x, y)
+          if metadata:
+            m = uss_metadata.USSMetadata(content)
+            removed_uvr = m.remove_uvr(uvr['message_id'])
+            if removed_uvr:
+              diff = uvrs.diff(removed_uvr, uvr)
+              if diff:
+                return self._format_status_code_to_jsend(
```

Comment:
Same comment as above, Delete should probably just be by ID and USS_ID.

Reply:
Deletion requires at least coordinates in addition to ```message_id```.  If the deleting client can store coordinate data about the UVR in addition to its ID, why not just store the entire UVR?  Requiring the full UVR definition, and requiring it to match, ensures that a UVR cannot be partially emplaced in the grid.